packages/cli/src/lib/upload/command-object.ts

```diff
@@ -1,10 +1,18 @@
-import { CommandModule } from 'yargs';
+import { ArgumentsCamelCase, CommandModule } from 'yargs';
 import { upload } from '@code-pushup/core';
+import { uploadConfigSchema } from '@code-pushup/models';
+import { CommandBase } from '../implementation/model';
 
 export function yargsUploadCommandObject() {
   return {
     command: 'upload',
     describe: 'Upload report results to the portal',
-    handler: upload as unknown as CommandModule['handler'],
+    handler: async <T>(args: ArgumentsCamelCase<T> & CommandBase) => {
+      const uploadOptions = {
+        ...args,
+        upload: uploadConfigSchema.parse(args.upload),
```

Comment:
Same question as for `autorun`.

Reply:
see upload command

Reply:
I see. That makes sense for the `upload` command, but it's not at all clear from the code.

How about just throwing an Error like this? I guess that would be more readable than whatever `parse` throws.

```ts
if (!args.upload) {
  throw new Error('Upload configuration not set');
}
// ... use args
```

Reply:
I can throw now and in the future we can add that validation to the command options